pkgdb2/api/admin.py

```diff
@@ -0,0 +1,355 @@
+    jsonout.status_code = httpcode
+    return jsonout
+
+
+@API.route('/admin/action/status', methods=['POST'])
+@is_admin
+def api_admin_action_edit_status():
+    '''
+Edit Admin Action status update
+-------------------------------
+    Edit the status of an Admin Action.
+
+    ::
+
+        /admin/action/status
+
+    Accept POST queries only.
+
+    :arg id: An integer representing the identifier of the admin
+        action to update in the database. The identifier is returned in the
+        API, see ``List admin actions``.
+    :arg status: The status to which the action should be updated.
+        Can be any of: ``Approved``, ``Awaiting Review``, ``Blocked``,
+        ``Denied``, ``Pending`` and ``Obsolete``.
+
+    ..note:: The ``Obsolete`` status can only be set by the person who made
+             the request.
+
+    Sample response:
+
+    ::
+
+        {
+          "output": "ok",
+          "messages": ["Admin action status updated to: Approved"]
+        }
+
+        {
+          "output": "notok",
+          "error": ["You are not an admin"]
+        }
+
+    '''
+    httpcode = 200
+    output = {}
+
+    action_status = pkgdblib.get_status(
+        SESSION, 'admin_status')['admin_status']
+
+    form = pkgdb2.forms.EditActionStatusForm(
+        csrf_enabled=False,
+        status=action_status,
+    )
+    if form.validate_on_submit():
+        action_id = form.id.data
+
+        admin_action = pkgdblib.get_admin_action(SESSION, action_id)
+        if not admin_action:
+            output['output'] = 'notok'
+            output['error'] = 'No Admin action with this identifier found'
+            httpcode = 500
+        else:
+
+            try:
+                message = pkgdblib.edit_action_status(
+                    SESSION,
+                    admin_action,
+                    action_status=form.status.data,
+                    user=flask.g.fas_user,
+                    message=form.message.data,
+                )
+                SESSION.commit()
+                output['output'] = 'ok'
+                output['messages'] = [message]
+            except pkgdblib.PkgdbException, err:  # pragma: no cover
+                # We can only reach here in two cases:
+                # 1) the user is not an admin, but that's taken care of
+                #    by the decorator
+                # 2) we have a SQLAlchemy problem when storing the info
+                #    in the DB which we cannot test
+                SESSION.rollback()
+                output['output'] = 'notok'
+                output['error'] = str(err)
+                httpcode = 500
+    else:
+        output['output'] = 'notok'
+        output['error'] = 'Invalid input submitted'
+        if form.errors:
+            detail = []
+            for error in form.errors:
+                detail.append('%s: %s' % (error,
+                              '; '.join(form.errors[error])))
+            output['error_detail'] = detail
+        httpcode = 500
```

Comment:
Same as for the 500, just trying to be consistent with the current mis-use of http error code